pkg/webhooks/setup.go

```diff
@@ -0,0 +1,110 @@
+package webhooks
+
+import (
+	"context"
+	hcov1beta1 "github.com/kubevirt/hyperconverged-cluster-operator/pkg/apis/hco/v1beta1"
+	"github.com/kubevirt/hyperconverged-cluster-operator/pkg/webhooks/mutator"
+	"github.com/kubevirt/hyperconverged-cluster-operator/pkg/webhooks/validator"
+	"os"
+	"path/filepath"
+
+	hcoutil "github.com/kubevirt/hyperconverged-cluster-operator/pkg/util"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+const (
+	webHookCertDirEnv = "WEBHOOK_CERT_DIR"
+)
+
+var (
+	logger = logf.Log.WithName("webhook-setup")
+)
+
+func GetWebhookCertDir() string {
+	webhookCertDir := os.Getenv(webHookCertDirEnv)
+	if webhookCertDir != "" {
+		return webhookCertDir
+	}
+
+	return hcoutil.DefaultWebhookCertDir
+}
+
+func SetupWebhookWithManager(ctx context.Context, mgr ctrl.Manager, isOpenshift bool) error {
+	operatorNsEnv, nserr := hcoutil.GetOperatorNamespaceFromEnv()
+	if nserr != nil {
+		logger.Error(nserr, "failed to get operator namespace from the environment")
+		return nserr
+	}
+
+	whHandler := validator.WebhookHandler{}
+	whHandler.Init(logger, mgr.GetClient(), operatorNsEnv, isOpenshift)
```

Comment:
Please replace with a single `NewWHValidator` function. There is no point to construct in two lines.